lib/features/mailbox/presentation/utils/mailbox_state_manager.dart

```diff
@@ -0,0 +1,44 @@
+
+import 'dart:collection';
+import 'package:core/utils/app_logger.dart';
+import 'package:jmap_dart_client/jmap/core/state.dart' as jmap;
+
+class MailboxStateManager {
+  final Queue<jmap.State> _stateQueue = Queue<jmap.State>();
+  bool _isProcessing = false;
+
+  void addState(jmap.State newState) {
```

Comment:
What happen if state is added during Queue manager is processing

Reply:
We use Queue & Completer to handle it